app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class CommentsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :check_comment, only: %i[edit update destroy]
+  before_action only: %i[update edit destroy] do
+    owner?(@new_comment.user)
+  end
+
+  def create
+    @post = Post.find(params[:comment][:post_id])
```

Comment:
but why do we need the user_id? we have the current user, we can build like so: `current_user.comments.build(comment_params)`
- let's build the private method `comment_params`, that way we can use the same params for update too.